daft/io/_iceberg.py

```diff
@@ -9,82 +9,52 @@
+    # from pyiceberg.table import Table as PyIcebergTable
+    import pyiceberg
 
 
 def _convert_iceberg_file_io_properties_to_io_config(props: Dict[str, Any]) -> Optional["IOConfig"]:
-    import pyiceberg
-    from packaging.version import parse
-    from pyiceberg.io import (
-        S3_ACCESS_KEY_ID,
-        S3_ENDPOINT,
-        S3_REGION,
-        S3_SECRET_ACCESS_KEY,
-        S3_SESSION_TOKEN,
-    )
-
+    """Property keys defined here: https://github.com/apache/iceberg-python/blob/main/pyiceberg/io/__init__.py."""
     from daft.io import AzureConfig, GCSConfig, IOConfig, S3Config
 
-    s3_mapping = {
-        S3_REGION: "region_name",
-        S3_ENDPOINT: "endpoint_url",
-        S3_ACCESS_KEY_ID: "key_id",
-        S3_SECRET_ACCESS_KEY: "access_key",
-        S3_SESSION_TOKEN: "session_token",
-    }
-    s3_args = dict()  # type: ignore
-    for pyiceberg_key, daft_key in s3_mapping.items():
-        value = props.get(pyiceberg_key, None)
-        if value is not None:
-            s3_args[daft_key] = value
-
-    if len(s3_args) > 0:
-        s3_config = S3Config(**s3_args)
-    else:
-        s3_config = None
-
-    gcs_config = None
-    azure_config = None
-    if parse(pyiceberg.__version__) >= parse("0.5.0"):
-        from pyiceberg.io import GCS_PROJECT_ID, GCS_TOKEN
-
-        gcs_mapping = {GCS_PROJECT_ID: "project_id", GCS_TOKEN: "token"}
-        gcs_args = dict()  # type: ignore
-        for pyiceberg_key, daft_key in gcs_mapping.items():
-            value = props.get(pyiceberg_key, None)
-            if value is not None:
-                gcs_args[daft_key] = value
-
-        if len(gcs_args) > 0:
-            gcs_config = GCSConfig(**gcs_args)
-
-        azure_mapping = {
-            "adlfs.account-name": "storage_account",
-            "adlfs.account-key": "access_key",
-            "adlfs.sas-token": "sas_token",
-            "adlfs.tenant-id": "tenant_id",
-            "adlfs.client-id": "client_id",
-            "adlfs.client-secret": "client_secret",
-        }
-
-        azure_args = dict()  # type: ignore
-        for pyiceberg_key, daft_key in azure_mapping.items():
-            value = props.get(pyiceberg_key, None)
-            if value is not None:
-                azure_args[daft_key] = value
-
-        if len(azure_args) > 0:
-            azure_config = AzureConfig(**azure_args)
-
-    if any([s3_config, gcs_config, azure_config]):
-        return IOConfig(s3=s3_config, gcs=gcs_config, azure=azure_config)
-    else:
+    any_props_set = False
+
+    def get_first_property_value(*property_names: str) -> Optional[Any]:
+        for property_name in property_names:
+            if property_value := props.get(property_name):
+                nonlocal any_props_set
+                any_props_set = True
+                return property_value
         return None
 
+    io_config = IOConfig(
+        s3=S3Config(
+            endpoint_url=props.get("s3.endpoint"),
```

Comment:
Should these (including the GCS ones below) also use `get_first_property_value` since we also want it to affect `any_props_set`?

Reply:
Ah good point, will change that